### ISSUE TITLE
Guest blog post with Mattermost?

### ISSUE THREAD

**Issue by jasonblais** [matterhorn-chat/matterhorn#244]
Hey everyone,

I found a blog post of your Matterhorn public release from a couple of weeks ago. It's fantastic! http://galois.com/blog/2017/05/matterhorn-experience-report/

We're working on a twitter announce (attached is a banner we prepared) and was wondering if you were interested on a guest blog post that we could add to https://about.mattermost.com/blog? We had one last year on [Sameroom integration](https://about.mattermost.com/sameroom-connects-mattermost-to-21-messengers-including-slack-skype-hipchat/) and are working on another one with Sourced Group. 

It could increase visibility for the integration and could be largely the same as http://galois.com/blog/2017/05/matterhorn-experience-report/ without many changes.

Alternatively we can link to your blog post as well, just something that came to mind.

Jason (from the Mattermost team)

![image](https://user-images.githubusercontent.com/13119842/26931217-b81da400-4c2d-11e7-8fde-b73a3028e021.png)

**Comment by jtdaugherty** [comment]
@jasonblais - thanks! This is great news. We'd love to be featured on your blog. The blog post we wrote has some material that would be appropriate for such a post, but the rest is really geared at Haskell programmers, not Mattermost users. So if we're going to have a blog post featured on your site then I'd like some time to write up a modified (and shorter) version pitched to the right audience. Are you on a particular timeline with this?

**Comment by jtdaugherty** [comment]
Also, I noticed that in the footer of that image above, it says "matterhorn by matterhorn-chat" - but since "matterhorn-chat" is the GitHub organization name and not a user, that reads strangely to me. Another option that might be more informative is `matterhorn by Galois, Inc.`. What do you think?

**Comment by jasonblais** [comment]
Good point on the screenshot, we can certainly update that.

No timeline on the blog post, I'd say what works the best for you and we can time the tweet and other marketing around it. We shouldn't do one close to the release (since people will be focused on the 3.10 release) so perhaps some time after the June 16th release?

Let me know what works for your team

**Comment by jtdaugherty** [comment]
I have two concerns about timeline right now:

* We'll need time to get the blog post ready
* When you announce this, it's very likely that people will download matterhorn expecting it to work with the latest Mattermost, which probably won't be the case, because we won't have had time to update matterhorn yet.

We tend to run one or Mattermost releases behind upstream in terms of compatibility, and although we're trying to do better about making it clear which Mattermost server version will work with our client, we still get occasional reports from users who have not noticed the versioning scheme.

One way to address my second concern is for us to note *very* clearly in the blog post that this only works with 3.8, and that users should get the appropriate matterhorn release for their server version.

So essentially the primary issue is that we'll need time to prepare this, and then it'll be on us to be clear in the messaging about compatibility.

**Comment by jasonblais** [comment]
Gotcha. I think it would be okay if your README and the blog post said it's compatible with 3.8. 

Our [React Native apps](https://github.com/mattermost/mattermost-mobile) also has a version compatibility check that confirms if the RN app version is compatible with the MM server version. Not sure if that could be useful here too.

I wouldn't worry about the timeline, we can of course make the guest blog post later too that works for your schedule.

**Comment by jasonblais** [comment]
On a related note, we were planning on including a section of the terminal client in the June 16th release note, as part of other new community integrations of the month.

If we included a clear note about compatibility (happy to work with you on it), would your team be okay with that?

Again, the guest post, etc. can be done later when you have the time,

**Comment by jtdaugherty** [comment]
Re: compatibility, we use a versioning scheme that encodes the compatible server version in the matterhorn version (e.g. `matterhorn-30802.1.0` means Mattermost server version `3.8.2`). Once we have a better way of sanity-checking that at startup (via the endpoint we requested), then it should be easier overall for users to get the client version right.

That's awesome about the June 16th release notes, and we would be grateful! We would be okay with a note about compatibility. The key point is this: get the right version of matterhorn for your server. If you upgrade your server, it's likely that upgrading Mattermost will require a matterhorn upgrade. Right now we support up to 3.8.

This weekend I'll work on the blog post content. Thanks for offering to do all this!

**Comment by jasonblais** [comment]
Perfect, I'll craft up the compatibility note and maybe share the release announce section here for review early next week.

Really excited for this project!

**Comment by jtdaugherty** [comment]
Thanks!

**Comment by jtdaugherty** [comment]
@jasonblais - what would be the best format to use for the content I'm writing? Markdown?

**Comment by jtdaugherty** [comment]
@jasonblais - I've written up a draft and will get it to you once our team has had a chance to sign off.

**Comment by jasonblais** [comment]
Awesome, that was quick!

Any formatting is fine actually, we convert the blog posts to HTML at the end.

**Comment by jtdaugherty** [comment]
Okay. We're nearing completion on our internal review and I've deliberately left out formatting, so you can introduce as much as you need once you get the text.

**Comment by jtdaugherty** [comment]
@jasonblais - once our draft is ready, what's the best way to get it to you?

**Comment by jasonblais** [comment]
You can either share it at https://pre-release.mattermost.com/ (my username is `jason`) or via email to jason[at]mattermost.com.

**Comment by jtdaugherty** [comment]
Okay. I'll e-mail it. Thanks! I'll send it in next couple of days.

**Comment by jasonblais** [comment]
Sweet, look forward to it!

**Comment by jtdaugherty** [comment]
@jasonblais - you should now have an e-mail with the blog post text.

**Comment by jasonblais** [comment]
@jtdaugherty Just finishing up the last pieces for 3.10 release announce..

To clarify on the correct terminology, is the https://github.com/matterhorn-chat/mattermost-api repository a "Haskell driver" or "Haskell library" for Mattermost?

**Comment by jtdaugherty** [comment]
It's a "Haskell library proving API bindings to Mattermost."

**Comment by jtdaugherty** [comment]
Providing, rather.

**Comment by jasonblais** [comment]
Got it, thanks!

**Comment by jasonblais** [comment]
We published the article, you can find it here: https://about.mattermost.com/mattermost-3-10/

We won't do our social media campaign until the Monday though, so feel free to let me know if there's anything to correct for your sections.

We're also in the process of scheduling next marketing topics, so I'll let you know soon when you can expect the Matterhorn guest blog post to be published and promoted.

**Comment by jtdaugherty** [comment]
That's fantastic coverage - thanks very much! Please remove the text about Windows support from your announcement. We have not had a chance to test that so I don't want to make that claim. (I also plan to remove it from our own documentation.)

**Comment by dagit** [comment]
I have actually built it using the Linux subsystem for Windows but not the
current release. I used that for awhile, but the terminal emulator I had to
use (based on putty) had redraw issues and eventually I abandoned it.

**Comment by jtdaugherty** [comment]
Thanks, Jason - that's good to know. If we can make regular windows testing a part of our work then I'd feel comfortable advertising and supporting it.

**Comment by jasonblais** [comment]
Thanks! I can take out Windows support when I'm back on my desk.

We could promote Windows support on a separate release announce too, when you feel comfortable advertising it

**Comment by jtdaugherty** [comment]
Since this is pending I'm going to go ahead and close. Thanks again, @jasonblais, for the opportunity!
